tests/www/approvals_views/test_prolongation.py

```diff
@@ -150,6 +150,60 @@ def test_prolong_approval_view_bad_reason(self):
             count=1,
         )
 
+    def test_prolongation_approval_view_with_disabled_choices(self):
+        """
+        Test the deactivation of reasons if too many prolongations have already been created.
+        """
+        # This should be several succeeding prolongations but this is good enough for our test
+        prolongation = ProlongationFactory(
+            approval=self.approval,
+            start_at=self.approval.end_at,
+            end_at=self.approval.end_at + timedelta(days=365 * 3 + 10),
+            reason=ProlongationReason.COMPLETE_TRAINING,
+        )
+        url = reverse("approvals:declare_prolongation", kwargs={"approval_id": self.approval.pk})
+
+        with freeze_time(prolongation.end_at):
+            self.client.force_login(self.employer)
+            response = self.client.get(url)
+            # Check the information card
+            soup = parse_response_to_soup(response, selector="div:has(> #disabledChoicesCollapseInfo)")
+            assert str(soup) == self.snapshot(name="missing_reason_info")
+            # Check the reason field
+            assert response.context["form"]["reason"].field.widget.disabled_choices == {"RQTH"}
+            assert {k for k, _v in response.context["form"]["reason"].field._choices} == {
```

Comment:
```suggestion
            assert set(response.context["form"]["reason"].field._choices) == {
```

Reply:
Mmh non, _choices n'est pas un dictionnaire.

Reply:
Effectivement... `for v, label in ...` alors ?

Reply:
Modifié